WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -3438,6 +3438,22 @@ public void onTrackableEvent(TrackableEvent event, Map<String, String> propertie
         mStoriesEventListener.onCancelSaveForMediaCollection(mediaFiles);
     }
 
+    @Override public void showPreview() {
+        PreviewLogicOperationResult opResult = mRemotePreviewLogicHelper.runPostPreviewLogic(
+                this,
+                mSite,
+                Objects.requireNonNull(mEditPostRepository.getPost()),
+                getEditPostActivityStrategyFunctions());
+        if (opResult == PreviewLogicOperationResult.MEDIA_UPLOAD_IN_PROGRESS
+            || opResult == PreviewLogicOperationResult.CANNOT_SAVE_EMPTY_DRAFT
+            || opResult == PreviewLogicOperationResult.CANNOT_REMOTE_AUTO_SAVE_EMPTY_POST
+        ) {
+            return;
+        } else if (opResult == PreviewLogicOperationResult.OPENING_PREVIEW) {
+            updatePostLoadingAndDialogState(PostLoadingState.PREVIEWING, mEditPostRepository.getPost());
+        }
+    }
```

Comment:
This logic has been extracted from the following lines, which is executed when tapping on the "Preview" option from the more settings dropdown:

https://github.com/wordpress-mobile/WordPress-Android/blob/3c3b35a69bbd3088c1fb09a9e9df7ae23d5de579/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java#L1425-L1438

Reply:
Ah, wdyt about reusing the new `showPreview()` function in line 1426 too, to avoid duplication? We can do that by perhaps returning a boolean in `showPreview()` and use that in line 1426 to return or not. Let me know if you'd like me to push a commit about that.

Reply:
Good idea! I thought about that but I noticed that on the `else if` block there's no `return` call, however, from what I see in the code it looks safe to add a `return` there because it won't prevent any later code to be executed, so I'll update this part 👍 .

Reply:
FWIW, my thinking is to replace the code block from line 1426 to line 1438 with:
```java
if (!showPreview()) {
    return false;
}
```

poivided we refactor the function to return a boolean.

Reply:
I've just applied your suggestion in [this commit](https://github.com/wordpress-mobile/WordPress-Android/pull/15001/commits/18eddc2507130a726f0061ad4273451a5febb1ef).